rest/firebaseQueryHandler.go

```diff
@@ -10,75 +10,75 @@ import (
 	"google.golang.org/api/iterator"
 )
 
-// getTracks gets a list of tracks from the DB
-func GetTracks(app *firebase.App, query models.FirebaseQuery) (data []models.IgcMetadata, err error) {
+// GetTracks gets a list of IgcMetadata from Firebase based on query
+func GetTracks(app *firebase.App, query models.FirebaseQuery) (data []models.IgcMetadata, err error){
 	ctx := context.Background()
 
 	client, err := app.Firestore(ctx)
 	if err != nil {
-		gtbackend.DebugLog(fileNameDB, "getTracks", err)
-
+		gtbackend.DebugLog(fileNameDB, "GetTracks", err)
 		return
 	}
 
-	println(query.Qt + " - " + query.UID)
-
-	if query.Qt == "Personal" {
+	// Start query to Firebase based on the Query Type
+	if query.Qt == "Private" {
 		iter := client.Collection(constant.IgcMetadata).
 			Where("UID", "==", query.UID).
 			OrderBy(query.Ord, query.OrdDir).Documents(ctx)
-		return processIterGetTracks(iter, query, false)
+		return processIterGetTracks(iter, query.Pg, "")
 	} else {
 		iter := client.Collection(constant.IgcMetadata).
 			Where("Privacy", "==", false).
 			OrderBy(query.Ord, query.OrdDir).Documents(ctx)
-		return processIterGetTracks(iter, query, true)
+		return processIterGetTracks(iter, query.Pg, query.UID)
 	}
 
 	return data, err
 }
 
-func processIterGetTracks(iter *firestore.DocumentIterator, query models.FirebaseQuery, filterSelf bool) (data []models.IgcMetadata, err error) {
-	total := 0
-	for ; total < (query.Pg-1)*constant.PageSize; total++ {
-		iter.Next()
-		if err == iterator.Done {
-			return data, err
-		}
-		if err != nil {
-			gtbackend.DebugLog(fileNameDB, "getTracks", err)
-			return data, err
-		}
-	}
+/**
+	Processes the request made to Firebase based
```

Comment:
Same comment as above, remember to write the name of the method first in a comment. Even though this is private